source/lambda/kendraIndexPoller/lambda_function.py

```diff
@@ -0,0 +1,34 @@
+import boto3
+
+def on_create(event, context):
+    kendra_client = boto3.client('kendra')
+    # get status of index
+    response = kendra_client.describe_index(Id=event['PhysicalResourceId'])
+    is_complete = False
+    status = response['Status']
+    if status == "CREATING":
+        print("still creating index")
+    elif status == "ACTIVE":
+        print("Kendra index is now active")
+        return { 'IsComplete': True }
+    elif status == "FAILED":
+        print("Kendra index creation has failed")
+        return { 'IsComplete': True }
+    elif status == "DELETING" or status == "SYSTEM_UPDATING":
+        print("Kendra index creation shows inconsistent status code, please fix and try again")
+    return { 'IsComplete': is_complete }
+
+def on_delete(event, context):
+    kendra_client = boto3.client('kendra')
+    response = kendra_client.describe_index(Id=event['PhysicalResourceId'])
+    if response['Status'] == "DELETING":
```

Comment:
You need to wait till the enitre Index is deleted. It could be in deleting state but may not get deleted for some error.

Reply:
There is no status called "DELETED", after the index is deleted, we can no longer call decribe_index on it, do you suggest checking for the kendra index id instead in this case?

Reply:
Yes, precisely